habitat-baselines/habitat_baselines/rl/ppo/trainer_agent.py

```diff
@@ -0,0 +1,283 @@
+from typing import TYPE_CHECKING, Any, Dict, Optional, Tuple
+
+import gym.spaces as spaces
+import numpy as np
+import torch
+from torch.optim.lr_scheduler import LambdaLR
+
+from habitat import logger
+from habitat_baselines.common.baseline_registry import baseline_registry
+from habitat_baselines.common.rollout_storage import (  # noqa: F401.
+    RolloutStorage,
+)
+from habitat_baselines.rl.ddppo.policy import (  # noqa: F401.
+    PointNavResNetNet,
+    PointNavResNetPolicy,
+)
+from habitat_baselines.rl.hrl.hierarchical_policy import (  # noqa: F401.
+    HierarchicalPolicy,
+)
+from habitat_baselines.rl.ppo import PPO  # noqa: F401.
+from habitat_baselines.rl.ppo.policy import Policy
+
+if TYPE_CHECKING:
+    from omegaconf import DictConfig
+
+
+class TrainerAgent:
```

Comment:
Can this be an attrs class with slots? It would simplify this logic a bit and potentially make it a bit more efficient.